src/core/fonts.js

```diff
@@ -1853,22 +1853,74 @@ class Font {
       sourceEnd,
       dest,
       destStart,
-      hintsValid
+      hintsValid,
+      fontXMin,
+      fontYMin,
+      fontXMax,
+      fontYMax
     ) {
       const glyphProfile = {
         length: 0,
         sizeOfInstructions: 0,
       };
-      if (sourceEnd - sourceStart <= 12) {
-        // glyph with data less than 12 is invalid one
+      if (
+        sourceStart < 0 ||
+        sourceStart >= source.length ||
+        sourceEnd >= source.length + 1 ||
```

Comment:
Couldn't this be a tiny bit shorter, if written follows?
```suggestion
        sourceEnd > source.length ||
```